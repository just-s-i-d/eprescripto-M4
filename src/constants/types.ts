export type InfoCardDetailsType =
  | {
      totalCount: number;
      title: string;
    }[]
  | undefined;

export type ErrorType = {
  message: string;
  type: string;
};
export type AppointmentDataType = {
  pId: string;
  email: string;
  lastVisited: string;
  avatar: string;
  gender: string;
  key: number;
  pName: string;
  timeSlot: string;
  contact: string;
  status: string;
  profilePic: string;
  referrer: string;
  date: string;
};

export type AppointmentsDataType = AppointmentDataType[] | undefined;

export type PrescriptionDataType = {
  key: React.Key;
  prescriptionId: string;
  name: string;
  duration: number;
  date: string;
  status: "Active" | "Completed" | "On Hold";
  notes: string;
};

export type PrescrtionsDataType = PrescriptionDataType[] | undefined;

export type ChartDataType = {
  labels: string[];
  data: number[];
  totalCount: number;
};

export type LineChartDataType = {
  labels: string[];
  data: number[];
};

export type LineChartApiResDataType = {
  seven: number[];
  ten: number[];
  thirty: number[];
};

export type ChartApiDataType = {
  pieChartData: ChartDataType;
  barChartData: ChartDataType;
  lineChartData: LineChartApiResDataType;
};

export type DoctorAppointmentsDataType = AppointmentDataType[];

export type ReviewDataType = {
  pId: string;
  pName: string;
  email: string;
  rating: 1 | 2 | 3 | 4 | 5;
  comment: string;
};
export type ReviewsDataType = ReviewDataType[];
