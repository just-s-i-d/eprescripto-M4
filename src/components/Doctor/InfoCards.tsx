import { useEffect, useState } from "react";
import { Row, Col, Card, Image, Typography, Skeleton } from "antd";

import ErrorBoundary from "@components/ErrorBoundary";
import CardTitle from "@components/ui/CardTitle";
import { ApiResponseData, InfoCardDetailsType } from "@constants/types";
import { CARD_PROPERTIES } from "@constants/constants";
import useStatesHook from "../../hooks/useStatesHook";

const { Text } = Typography;

type InfoCardsProps = {
  cardDetails?: ApiResponseData<InfoCardDetailsType>;
  infoCards: ReturnType<
    typeof useStatesHook<ApiResponseData<InfoCardDetailsType>>
  >;
};

const getBgColor = (index: number) => {
  switch (index + 1) {
    case 1:
      return "bg-infoCardColors-color1";
    case 2:
      return "bg-infoCardColors-color2";
    case 3:
      return "bg-infoCardColors-color3";
    case 4:
      return "bg-infoCardColors-color4";
  }
};
const InfoCards = ({ cardDetails, infoCards }: InfoCardsProps) => {
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    cardDetails && setTimeout(() => setLoading(false), 500);
  }, [cardDetails]);
  return (
    <Card
      className="info-cards-body w-full h-[22vh] max-md:h-full flex justify-center"
      bordered={false}
    >
      <ErrorBoundary
        error={infoCards.error}
        refreshComponent={() => infoCards.setRefresh((prev) => !prev)}
      >
        <Row
          gutter={30}
          className="w-full max-md:flex max-md:flex-wrap gap-y-6 justify-center max-lg:flex-nowrap"
        >
          {cardDetails?.map((element, index) => {
            return (
              <ErrorBoundary key={element.attributes.title}>
                <Col className="basis-3/12 max-md:basis-6/12 max-sm:basis-9/12">
                  <Card
                    className={`info-card h-[17vh] py-8 px-4 max-sm:py-5 max-w-[390px] max-xl:h-[16vh] max-xl:min-w-[135px] max-sm:h-[150px] ${getBgColor(
                      index,
                    )} text-white`}
                    bordered={false}
                  >
                    <div>
                      <div className={`flex items-center gap-4 mb-2 mt-2`}>
                        {loading ? (
                          <Skeleton.Avatar active shape="square" size={50} />
                        ) : (
                          <Image
                            src={CARD_PROPERTIES[index]}
                            alt="Icon image"
                            className="w-[50px] max-xxl:w-[35px] max-md:w-[40px]"
                            preview={false}
                          />
                        )}
                        {loading ? (
                          <Skeleton.Button />
                        ) : (
                          <Text className="text-4xl max-xl:text-3xl text-white max-md:text-4xl">
                            {element.attributes.totalCount}
                          </Text>
                        )}
                      </div>
                      {loading ? (
                        <Skeleton.Input active />
                      ) : (
                        <CardTitle className="max-xxl:text-lg max-xl:text-base max-md:text-lg">
                          {element.attributes.title}
                        </CardTitle>
                      )}
                    </div>
                  </Card>
                </Col>
              </ErrorBoundary>
            );
          })}
        </Row>
      </ErrorBoundary>
    </Card>
  );
};
export default InfoCards;
