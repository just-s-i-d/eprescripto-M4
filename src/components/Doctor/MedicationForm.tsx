import { Input, Select, Form } from "antd";
import { CloseOutlined } from "@ant-design/icons";
import { MedicationType } from "@constants/types";

const { Item } = Form;
type MedicationFormPropsType = {
  fields: MedicationType;
  removeable?: boolean;
  removeMedicationInput: (id: number) => void;
};
const MedicationForm = ({
  fields,
  removeMedicationInput,
  removeable,
}: MedicationFormPropsType) => {
  return (
    <>
      <div className="mt-4 mb-2 flex justify-between">
        <h2 className="text-xl">Medication</h2>
        {removeable && (
          <CloseOutlined
            onClick={() => removeMedicationInput(fields.id)}
            data-testid="remove-field"
            className="text-[16px]"
          />
        )}
      </div>

      <div className="w-full grid sm:grid-cols-2 md:grid-cols-4 gap-x-5 gap-y-3">
        <Item
          name={fields.medicine}
          label="Medicine Name"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[a-zA-Z0-9.-]+$/,
              message: "Special characters not allowed",
            },
          ]}
        >
          <Input />
        </Item>
        <Item
          name={fields.dosage}
          label="Dosage"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[0-9/mg]/,
              message: "Enter a valid dosage",
            },
          ]}
        >
          <Input />
        </Item>
        <Item
          name={fields.timesPerDay}
          label="Frequency"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
          ]}
        >
          <Select
            options={[
              {
                value: 1,
                label: "1/Day",
              },
              {
                value: 2,
                label: "2/Day",
              },
              {
                value: 3,
                label: "3/Day",
              },
            ]}
          />
        </Item>
        <div className="flex gap-6">
          <Item
            name={fields.instruction}
            className="w-full"
            label="Instruction"
            required
            rules={[
              {
                required: true,
                message: "Field cannot be empty",
              },
            ]}
          >
            <Select
              options={[
                {
                  value: "beforeFood",
                  label: "Before eating food",
                },
                {
                  value: "afterFood",
                  label: "After eating Food",
                },
              ]}
            />
          </Item>
        </div>
      </div>
    </>
  );
};
export default MedicationForm;
