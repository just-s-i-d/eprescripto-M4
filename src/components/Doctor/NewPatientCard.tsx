import ErrorBoundary from "@components/ErrorBoundary";
import { Form, FormInstance, Input, Select } from "antd";
import FormItem from "antd/es/form/FormItem";
import TextArea from "antd/es/input/TextArea";

type NewPatientCardPropType = {
  newPatientForm: FormInstance;
};
const NewPatientCard = ({ newPatientForm }: NewPatientCardPropType) => {
  return (
    <ErrorBoundary>
      <Form form={newPatientForm} layout="vertical" className="mt-6 px-4">
        <FormItem
          name="pName"
          label="Patient Name"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[A-Za-z\s]+$/,
              message: "Special characters not allowed",
            },
          ]}
        >
          <Input />
        </FormItem>
        <FormItem
          name="email"
          label="Email"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[\w.-]+@[\w.-]+\.\w+$/,
              message: "Enter a valid mail",
            },
          ]}
        >
          <Input />
        </FormItem>
        <div className="flex gap-6">
          <FormItem
            name="dob"
            label="Date of Birth"
            required
            className="w-full"
            rules={[
              {
                required: true,
                message: "Please enter your age!",
              },
            ]}
          >
            <Input type="date" />
          </FormItem>
          <FormItem
            name="gender"
            label="Gender"
            required
            className="w-full"
            rules={[
              {
                required: true,
                message: "Please select your gender!",
              },
            ]}
          >
            <Select
              placeholder="Select your gender"
              options={[
                { value: "male", label: "Male" },
                { value: "female", label: "Female" },
                { value: "other", label: "Other" },
              ]}
            />
          </FormItem>
        </div>
        <FormItem
          name="contactNo"
          label="Mobile"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[6-9]\d{9}$/,
              message: "Enter valid no.",
            },
          ]}
        >
          <Input />
        </FormItem>
        <FormItem name="allergies" label="Mention allergies , if any">
          <TextArea placeholder="Mention your allergies here" />
        </FormItem>
      </Form>
    </ErrorBoundary>
  );
};
export default NewPatientCard;
