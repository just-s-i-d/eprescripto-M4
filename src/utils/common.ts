import { notification } from "antd";

export const showToast = (message: string, type: "success" | "error") => {
  notification.open({
    message: type === "success" ? "Success" : "Error",
    description: message,
    type: type,
    className: `bg-${type}-bgColor border rounded border-${type}-borderColor`,
    duration: 3,
  });
};
