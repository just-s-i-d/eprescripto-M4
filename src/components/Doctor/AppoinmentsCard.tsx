import { useState } from "react";
import { Card, Skeleton, Avatar, Tooltip, Image } from "antd";
import { CloseCircleTwoTone, PhoneTwoTone } from "@ant-design/icons";

import noDataPng from "@assets/no-data.png";
import CardTitle from "@components/ui/CardTitle";
import PopModal from "@components/ui/PopModal";
import profilePic from "@assets/profilepic.png";
import {
  ApiResponseData,
  ApiResponseDataType,
  AppointmentDataType,
} from "@constants/types";
import { cancelAppointment } from "@utils/Doctor";
import ErrorBoundary from "@components/ErrorBoundary";
import useStatesHook from "src/hooks/useStatesHook";
import { showToast } from "@utils/common";

type AppointmentCardPropsType = {
  appointments?: ApiResponseData<AppointmentDataType>;
  loading: boolean;
  appointmentsState: ReturnType<
    typeof useStatesHook<ApiResponseData<AppointmentDataType>>
  >;
  getDataForAppointments: () => void;
};

const AppoinmentsCard = ({
  appointments,
  loading,
  appointmentsState,
  getDataForAppointments,
}: AppointmentCardPropsType) => {
  const [open, setOpen] = useState<boolean>(false);
  const [selectedPatient, setSelectedPatient] =
    useState<ApiResponseDataType<AppointmentDataType>>();
  const handlerCancleAppointment = (
    patient: ApiResponseDataType<AppointmentDataType>,
  ) => {
    setOpen(true);
    setSelectedPatient(patient);
  };
  const confirmHandler = async () => {
    if (selectedPatient) {
      cancelAppointment(selectedPatient)
        .then((res) => {
          showToast(res.message, res.type);
          getDataForAppointments();
        })
        .catch((error) => {
          showToast(error.message, error.type);
        });
    }
  };
  return (
    <>
      <Card
        className={`w-[32%] max-xxl:h-[59vh] max-h-[59vh] max-xl:max-h-[60vh] max-xxl:w-[48%] max-xl:w-[48%] max-md:w-9/12 max-sm:w-full `}
        bordered={false}
      >
        <ErrorBoundary
          refreshComponent={() => appointmentsState.setRefresh((prev) => !prev)}
          error={appointmentsState.error}
        >
          <CardTitle>Today's Appointments</CardTitle>
          {!appointmentsState.loading && !appointmentsState.data?.length ? (
            <div className="flex flex-col items-center justify-center">
              <Image src={noDataPng} preview={false} />
              <h2 className="text-lg mt-12 text-center">No Appointments</h2>
            </div>
          ) : (
            <div className="mx-auto px-4 h-[44vh] overflow-y-auto">
              <ul className="mt-1 divide-y">
                {appointments?.map((item, idx) => (
                  <li
                    key={idx}
                    className="py-4 mb-2 px-2 bg-[#F0F3F4] rounded-lg"
                  >
                    <div className="flex items-start justify-between">
                      <div className="flex gap-3">
                        {loading ? (
                          <Skeleton.Avatar active size={48} />
                        ) : (
                          <Avatar
                            size={48}
                            src={item.attributes?.profilePic || profilePic}
                          />
                        )}
                        <Skeleton
                          loading={loading}
                          active
                          avatar={false}
                          title={false}
                          paragraph={{ rows: 2, width: [100, 100] }}
                        >
                          <div>
                            <span className="block text-base text-gray-700">
                              {item.attributes?.pName}
                            </span>
                            <span className="block text-[12px] text-gray-600">
                              {item.attributes?.timeSlot}
                            </span>
                          </div>
                        </Skeleton>
                      </div>
                      {loading ? (
                        <Skeleton.Button active />
                      ) : (
                        <span
                          className={`py-1 px-2 rounded-lg ${item.attributes?.status.toLowerCase()}`}
                        >
                          {item.attributes?.status}
                        </span>
                      )}
                    </div>
                    <div className="flex justify-between">
                      <span className="block text-sm text-gray-700 font-bold">
                        <span className="mr-1 py-1 text-gray-500 font-semibold">
                          Referrer
                        </span>
                        {item.attributes.referrer || "NA"}
                      </span>
                      <div className="flex justify-end gap-4 text-lg">
                        <Tooltip title={item.attributes?.contact}>
                          <PhoneTwoTone
                            className="rotate-90 cursor-pointer"
                            href={`tel:${item.attributes?.contact}`}
                          />
                        </Tooltip>

                        {item.attributes.status === "Pending" && (
                          <CloseCircleTwoTone
                            className="cursor-pointer"
                            onClick={() => handlerCancleAppointment(item)}
                          />
                        )}
                      </div>
                      <PopModal
                        setOpen={setOpen}
                        open={open}
                        okButtonText="Sure"
                        title="Cancel Appointment"
                        footer={true}
                        confirmHandler={confirmHandler}
                      >
                        Are you sure,you want to cancel the appoinment for{" "}
                        {selectedPatient?.attributes.pName}?
                      </PopModal>
                    </div>
                  </li>
                ))}
              </ul>
            </div>
          )}
        </ErrorBoundary>
      </Card>
    </>
  );
};

export default AppoinmentsCard;
