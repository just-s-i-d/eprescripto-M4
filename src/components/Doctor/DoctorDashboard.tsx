import { useEffect, useState } from "react";

import NextPatientCard from "./NextPatientCard";
import InfoCards from "./InfoCards";
import ErrorBoundary from "@components/ErrorBoundary";
import PieChart from "../Charts/PieChart";
import LineChart from "@components/Charts/LineChart";
import BarChart from "@components/Charts/BarChart";
import AppoinmentsCard from "./AppoinmentsCard";
import {
  AppointmentsDataType,
  ChartDataType,
  InfoCardDetailsType,
  LineChartApiResDataType,
} from "@constants/types";
import {
  barChartDataEndPoint,
  getAppointmentData,
  getChartData,
  getDataForLineGraph,
  getInfoCardsData,
  lineChartDataEndPoint,
  pieChartDataEndPoint,
} from "@utils/Doctor";

const DoctorDashboard: React.FC = () => {
  const [lineChartData, setLineChartData] = useState<LineChartApiResDataType>();
  const [infoCardError, setInfoCardError] = useState(false);
  const [appointmentCardError, setAppointmentCardError] = useState(false);
  const [appointments, setAppointments] = useState<AppointmentsDataType>();
  const [infoCardsData, setInfoCardsData] = useState<InfoCardDetailsType>();
  const [pieChartData, setPieChartData] = useState<ChartDataType>();
  const [barChartData, setBarChartData] = useState<ChartDataType>();

  useEffect(() => {
    getInfoCardsData()
      .then((res) => {
        setInfoCardsData(res);
        setInfoCardError(false);
      })
      .catch(() => setInfoCardError(true));
    getAppointmentData()
      .then((res) => setAppointments(res))
      .catch(() => setAppointmentCardError(true));
    getChartData(pieChartDataEndPoint).then((res) => {
      setPieChartData(res.pieChartData);
    });
    getChartData(lineChartDataEndPoint).then((res) => {
      setLineChartData(res.lineChartData);
    });
    getChartData(barChartDataEndPoint).then((res) => {
      setBarChartData(res.barChartData);
    });
  }, []);
  return (
    <>
      <ErrorBoundary>
        <div className="max-w-full flex">
          <div className="w-full flex flex-col gap-6">
            <ErrorBoundary error={infoCardError}>
              <InfoCards cardDetails={infoCardsData} error={infoCardError} />
            </ErrorBoundary>
            <div className="flex flex-wrap justify-between gap-y-6 max-tablet:justify-center">
              <ErrorBoundary>
                <NextPatientCard
                  appointment={appointments && appointments[0]}
                  error={appointmentCardError}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <PieChart
                  chartData={pieChartData}
                  title="Ratings Composition"
                  label="Reviews"
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <AppoinmentsCard
                  appointments={appointments}
                  loading={appointmentCardError}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <LineChart
                  chartData={lineChartData}
                  title="Patients in Last 7 Days"
                  xAxesTitle="Dates"
                  yAxesTitle="Number of Patients"
                  getDataForLineGraph={getDataForLineGraph}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <BarChart
                  chartData={barChartData}
                  title="Age Groups Comparison"
                  label="Patients"
                  xAxesTitle="Different Age Groups (In Years)"
                  yAxesTitle="Number of Patients"
                />
              </ErrorBoundary>
            </div>
          </div>
        </div>
      </ErrorBoundary>
    </>
  );
};

export default DoctorDashboard;
