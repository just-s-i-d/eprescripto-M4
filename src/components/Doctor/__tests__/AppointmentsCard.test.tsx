import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import AppoinmentsCard from "../AppoinmentsCard";
import useStatesHook from "@hooks/useStatesHook";
import { cancelAppointment, getData } from "@utils/Doctor";
import { ApiResponseData, AppointmentDataType } from "@constants/types";
import { showToast } from "@utils/common";
import "@test/matchMedia";
jest.mock("@utils/Doctor", () => ({
  getData: jest.fn().mockResolvedValue([
    {
      id: 8,
      attributes: {
        pName: "John Doe",
        timeSlot: "10:00 AM",
        status: "Pending",
        referrer: "Dr. Smith",
        contact: "123-456-7890",
      },
    },
  ]),
  cancelAppointment: jest
    .fn()
    .mockResolvedValue({ message: "Cancelled", type: "success" }),
}));
jest.mock("@utils/common", () => ({
  showToast: jest.fn(),
}));
jest.mock("@hooks/useStatesHook");

describe("AppoinmentsCard component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("Test to check if Appointments cards renders correctly with data", async () => {
    const mockedData: ApiResponseData<AppointmentDataType> =
      await getData("Mocked data");
    (useStatesHook as jest.Mock).mockReturnValue({
      data: mockedData,
      loading: false,
    });
    const mockUseStatesHook =
      useStatesHook<ApiResponseData<AppointmentDataType>>();
    const { asFragment } = render(
      <AppoinmentsCard
        appointments={mockedData}
        loading={false}
        appointmentsState={mockUseStatesHook}
        getDataForAppointments={() => {}}
      />,
    );
    await waitFor(() => {
      expect(
        screen.getByText(mockedData[0].attributes.pName),
      ).toBeInTheDocument();
      expect(
        screen.getByText(mockedData[0].attributes.timeSlot),
      ).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders no data state", async () => {
    (useStatesHook as jest.Mock).mockReturnValue({ data: [] });
    const mockUseStatesHook =
      useStatesHook<ApiResponseData<AppointmentDataType>>();
    render(
      <AppoinmentsCard
        appointments={[]}
        loading={false}
        appointmentsState={mockUseStatesHook}
        getDataForAppointments={() => {}}
      />,
    );
    const noDataState = screen.getByText("No Appointments");
    expect(noDataState).toBeInTheDocument();
  });
  it("handling error state", async () => {
    (useStatesHook as jest.Mock).mockReturnValue({ data: [] });
    const mockUseStatesHook =
      useStatesHook<ApiResponseData<AppointmentDataType>>();
    const setRefreshMock = jest.fn();
    render(
      <AppoinmentsCard
        appointments={[]}
        loading={false}
        appointmentsState={{
          ...mockUseStatesHook,
          setRefresh: setRefreshMock,
          error: true,
        }}
        getDataForAppointments={() => {}}
      />,
    );
    const tryAgainBtn = screen.getByText("Try again");
    fireEvent.click(tryAgainBtn);
    expect(setRefreshMock).toHaveBeenCalled();
  });

  it("Cancel appointment successfull and show toast test", async () => {
    const mockedData: ApiResponseData<AppointmentDataType> =
      await getData("Mocked data");
    (useStatesHook as jest.Mock).mockReturnValue({
      data: mockedData,
      loading: false,
    });
    const mockUseStatesHook =
      useStatesHook<ApiResponseData<AppointmentDataType>>();
    render(
      <AppoinmentsCard
        appointments={mockedData}
        loading={false}
        appointmentsState={mockUseStatesHook}
        getDataForAppointments={() => {}}
      />,
    );
    const cancelBtn = screen.getByTestId("cancel-appointment-btn");
    expect(cancelBtn).toBeInTheDocument();
    fireEvent.click(cancelBtn);
    expect(screen.getByText("Cancel Appointment")).toBeInTheDocument();
    const confirmBtn = screen.getByText("Sure");
    fireEvent.click(confirmBtn);
    await waitFor(() => {
      expect(cancelAppointment).toHaveBeenCalledWith(mockedData[0]);
      expect(showToast).toHaveBeenCalledWith("Cancelled", "success");
    });
  });
  it("Cancel appointment Failure and show toast test", async () => {
    (cancelAppointment as jest.Mock).mockRejectedValue({
      message: "Error",
      type: "error",
    });
    const mockedData: ApiResponseData<AppointmentDataType> =
      await getData("Mocked data");
    (useStatesHook as jest.Mock).mockReturnValue({
      data: mockedData,
      loading: false,
    });
    const mockUseStatesHook =
      useStatesHook<ApiResponseData<AppointmentDataType>>();
    render(
      <AppoinmentsCard
        appointments={mockedData}
        loading={false}
        appointmentsState={mockUseStatesHook}
        getDataForAppointments={() => {}}
      />,
    );
    const cancelBtn = screen.getByTestId("cancel-appointment-btn");
    expect(cancelBtn).toBeInTheDocument();
    fireEvent.click(cancelBtn);
    expect(screen.getByText("Cancel Appointment")).toBeInTheDocument();
    const confirmBtn = screen.getByText("Sure");
    fireEvent.click(confirmBtn);
    await waitFor(() => {
      expect(cancelAppointment).toHaveBeenCalledWith(mockedData[0]);
      expect(showToast).toHaveBeenCalledWith("Error", "error");
    });
  });
});
