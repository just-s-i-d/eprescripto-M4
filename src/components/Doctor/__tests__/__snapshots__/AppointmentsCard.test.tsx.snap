// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppoinmentsCard component Test to check if Appointments cards renders correctly with data 1`] = `
<DocumentFragment>
  <div
    class="ant-card w-[32%] max-xxl:h-[59vh] max-h-[59vh] max-xl:max-h-[60vh] max-xxl:w-[48%] max-xl:w-[48%] max-md:w-9/12 max-sm:w-full  css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-card-body"
    >
      <h1
        class="text-xl font-bold mb-6 undefined"
      >
        Today's Appointments
      </h1>
      <div
        class="mx-auto px-4 h-[44vh] overflow-y-auto"
      >
        <ul
          class="mt-1 divide-y"
        >
          <li
            class="py-4 mb-2 px-2 bg-[#F0F3F4] rounded-lg"
          >
            <div
              class="flex items-start justify-between"
            >
              <div
                class="flex gap-3"
              >
                <span
                  class="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-1adbn6x"
                  style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
                >
                  <img
                    src="profilepic.png"
                  />
                </span>
                <div>
                  <span
                    class="block text-base text-gray-700"
                  >
                    John Doe
                  </span>
                  <span
                    class="block text-[12px] text-gray-600"
                  >
                    10:00 AM
                  </span>
                </div>
              </div>
              <span
                class="py-1 px-2 rounded-lg pending"
              >
                Pending
              </span>
            </div>
            <div
              class="flex justify-between"
            >
              <span
                class="block text-sm text-gray-700 font-bold"
              >
                <span
                  class="mr-1 py-1 text-gray-500 font-semibold"
                >
                  Referrer
                </span>
                Dr. Smith
              </span>
              <div
                class="flex justify-end gap-4 text-lg"
              >
                <span
                  aria-label="phone"
                  class="anticon anticon-phone rotate-90 cursor-pointer"
                  href="tel:123-456-7890"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="phone"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M721.7 184.9L610.9 295.8l120.8 120.7-8 21.6A481.29 481.29 0 01438 723.9l-21.6 8-.9-.9-119.8-120-110.8 110.9 104.5 104.5c10.8 10.7 26 15.7 40.8 13.2 117.9-19.5 235.4-82.9 330.9-178.4s158.9-213.1 178.4-331c2.5-14.8-2.5-30-13.3-40.8L721.7 184.9z"
                      fill="#e6f4ff"
                    />
                    <path
                      d="M877.1 238.7L770.6 132.3c-13-13-30.4-20.3-48.8-20.3s-35.8 7.2-48.8 20.3L558.3 246.8c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l89.6 89.7a405.46 405.46 0 01-86.4 127.3c-36.7 36.9-79.6 66-127.2 86.6l-89.6-89.7c-13-13-30.4-20.3-48.8-20.3a68.2 68.2 0 00-48.8 20.3L132.3 673c-13 13-20.3 30.5-20.3 48.9 0 18.5 7.2 35.8 20.3 48.9l106.4 106.4c22.2 22.2 52.8 34.9 84.2 34.9 6.5 0 12.8-.5 19.2-1.6 132.4-21.8 263.8-92.3 369.9-198.3C818 606 888.4 474.6 910.4 342.1c6.3-37.6-6.3-76.3-33.3-103.4zm-37.6 91.5c-19.5 117.9-82.9 235.5-178.4 331s-213 158.9-330.9 178.4c-14.8 2.5-30-2.5-40.8-13.2L184.9 721.9 295.7 611l119.8 120 .9.9 21.6-8a481.29 481.29 0 00285.7-285.8l8-21.6-120.8-120.7 110.8-110.9 104.5 104.5c10.8 10.8 15.8 26 13.3 40.8z"
                      fill="#1677ff"
                    />
                  </svg>
                </span>
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle cursor-pointer"
                  data-testid="cancel-appointment-btn"
                  role="img"
                  tabindex="-1"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      fill="#1677ff"
                    />
                    <path
                      d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm171.8 527.1c1.2 1.5 1.9 3.3 1.9 5.2 0 4.5-3.6 8-8 8l-66-.3-99.3-118.4-99.3 118.5-66.1.3c-4.4 0-8-3.6-8-8 0-1.9.7-3.7 1.9-5.2L471 512.3l-130.1-155a8.32 8.32 0 01-1.9-5.2c0-4.5 3.6-8 8-8l66.1.3 99.3 118.4 99.4-118.5 66-.3c4.4 0 8 3.6 8 8 0 1.9-.6 3.8-1.8 5.2l-130.1 155 129.9 154.9z"
                      fill="#e6f4ff"
                    />
                    <path
                      d="M685.8 352c0-4.4-3.6-8-8-8l-66 .3-99.4 118.5-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155-130.1 154.9a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3 99.3-118.5L611.7 680l66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.9 512.2l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
                      fill="#1677ff"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
